sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -143,14 +145,17 @@ object DataSourceV2Strategy extends Strategy {
         DataSourceV2ScanExec(
           r.output, r.source, r.options, r.pushedFilters, r.readSupport, scanConfig)) :: Nil
 
-    case WriteToDataSourceV2(writer, query) =>
-      WriteToDataSourceV2Exec(writer, planLater(query)) :: Nil
+    case WriteToDataSourceV2(writer, writeConfig, query) =>
+      WriteToDataSourceV2Exec(writer, writeConfig, planLater(query)) :: Nil
 
     case AppendData(r: DataSourceV2Relation, query, _) =>
-      WriteToDataSourceV2Exec(r.newWriteSupport(), planLater(query)) :: Nil
+      val writeSupport = r.newWriteSupport()
+      val writeConfig = writeSupport.createWriteConfig(
+        query.schema, new DataSourceOptions(r.options.asJava))
+      WriteToDataSourceV2Exec(r.newWriteSupport(), writeConfig, planLater(query)) :: Nil
```

Comment:
Why do `r.newWriteSupport()` again? We can reuse `writeSupport`.

Reply:
Looks like an accident to me.